x-pack/plugins/observability_solution/observability_ai_assistant/common/utils/with_token_budget.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { encode } from 'gpt-tokenizer';
+
+export function withTokenBudget<T extends unknown>(
```

Comment:
@miltonhultgren I'm a little late to the party but it would be nice with a unit test for this function

Reply:
Totally agree, it's on my list for Part 2!